src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/repositories/ResourceContainerRepository.kt

```diff
@@ -0,0 +1,154 @@
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.ResourceMetadataMapper
+import org.wycliffeassociates.resourcecontainer.ResourceContainer
+
+class ResourceContainerRepository(
+        private val database: AppDatabase
+) : IResourceContainerRepository {
+    private val collectionDao = database.getCollectionDao()
+    private val contentDao = database.getContentDao()
+    private val resourceMetadataDao = database.getResourceMetadataDao()
+    private val languageDao = database.getLanguageDao()
+    private val resourceLinkDao = database.getResourceLinkDao()
+
+    override fun importResourceContainer(rc: ResourceContainer, rcTree: Tree, languageSlug: String): Completable {
+        return Completable.fromAction {
+            database.transaction { dsl ->
+                val language = LanguageMapper().mapFromEntity(languageDao.fetchBySlug(languageSlug, dsl))
+                val metadata = rc.manifest.dublinCore.mapToMetadata(rc.dir, language)
+                val dublinCoreFk = resourceMetadataDao.insert(ResourceMetadataMapper().mapToEntity(metadata), dsl)
+
+                // TODO: Should this only happen if we are importing a "help" RC? What if the help is imported first,
+                // TODO: ... then another related bundle RC is imported?
+                val relatedDublinCoreIds: List<Int> =
+                        linkRelatedResourceContainers(dublinCoreFk, rc.manifest.dublinCore.relation, dsl)
+
+                // TODO: Probably shouldn't hardcode "help"
+                if (rc.type() == "help") {
+                    relatedDublinCoreIds.forEach { relatedId ->
+                        val ih = ImportHelper(dublinCoreFk, relatedId, dsl)
+                        ih.importCollection(null, rcTree)
+                    }
+                } else {
+                    val ih = ImportHelper(dublinCoreFk, null, dsl)
+                    ih.importCollection(null, rcTree)
+                }
+            }
+        }.subscribeOn(Schedulers.io())
+    }
+
+    private fun linkRelatedResourceContainers(
+            dublinCoreFk: Int,
+            relations: List<String>,
+            dsl: DSLContext)
+            : List<Int> {
+        val relatedIds = mutableListOf<Int>()
+        relations.forEach { relation ->
+            val parts = relation.split('/')
+            resourceMetadataDao.fetchByLanguageAndIdentifier(parts[0], parts[1], dsl)
+                    ?.let { relatedDublinCore ->
+                        // TODO: Only add link if it doesn't exist already
+                        resourceMetadataDao.addLink(dublinCoreFk, relatedDublinCore.id, dsl)
+                        relatedIds.add(relatedDublinCore.id)
+                    }
+        }
+        return relatedIds
+    }
+
+    inner class ImportHelper(val dublinCoreId: Int, val relatedBundleDublinCoreId: Int?, val dsl: DSLContext, var collectionsImported: Int = 0) {
```

Comment:
```
inner class ImportHelper(
    val dublinCoreId: Int,
    val relatedBundleDublinCoreId: Int?, 
    val dsl: DSLContext, 
    var collectionsImported: Int = 0
) {
```

(one reason for line length limits is that it keeps you from having to scroll horizontally when looking at code on github lol)